### PR TITLE
Fix compatibility test

### PR DESCRIPTION
**Pull request by MDunitz**
## Need
- to have working compatibility tests so when they fail we are given meaningful information

## Approach
- require anndata >= 0.7
Scanpy also has the requirement.
- added a version check/warning to ensure users are aware
- updated versions used in compatibility testing

### CODE REVIEW

File: server/requirements.txt [hunk 1]
```diff
@@ -1,4 +1,4 @@
-anndata>=0.6.20
+anndata>=0.7
```
Comment (maniarathi) [inline server/requirements.txt:1]: Not sure if this is legit or not, but does this need to be the full semvar in order for the comparison to work? i.e. 0.7.0?

Comment (MDunitz) [reply server/requirements.txt:1]: it worked locally but I can update it just to be safe

Comment (MDunitz) [inline server/requirements.txt:1]:
```suggestion
anndata>=0.7.0
```

File: .github/workflows/compatibility_tests.yml [hunk 2]
```diff
@@ -29,7 +29,7 @@ jobs:
     strategy:
       matrix:
         python-version: [3.6, 3.7, 3.8]
-        anndata-version: [0.6.22.post1, 0.7.1]
+        anndata-version: [0.7, 0.7.1, 0.7.2, 0.7.3, 0.7.4]
```
Comment (MDunitz) [inline .github/workflows/compatibility_tests.yml:32]:
```suggestion
        anndata-version: [0.7.0, 0.7.1, 0.7.2, 0.7.3, 0.7.4]
```

File: server/data_anndata/anndata_adaptor.py [hunk 3]
```diff
@@ -177,10 +177,10 @@ def _load_data(self, data_locator):
             )
 
     def _validate_and_initialize(self):
-        if anndata_version_is_pre_070() and self.server_config.adaptor__anndata_adaptor__backed:
+        if anndata_version_is_pre_070():
             warnings.warn(
-                "Use of --backed mode with anndata versions older than 0.7 will have serious "
-                "performance issues. Please update to at least anndata 0.7 or later."
+                "Use of anndata versions older than 0.7 will have serious "
+                "issues. Please update to at least anndata 0.7 or later."
```
Comment (maniarathi) [inline server/data_anndata/anndata_adaptor.py:183]: Nit: I think there's a weird line break -- more words can go on line 182 :P

**Review by maniarathi — APPROVED** [review]
LGTM, thanks!

File: server/data_anndata/anndata_adaptor.py [hunk 4]
```diff
@@ -177,10 +177,10 @@ def _load_data(self, data_locator):
             )
 
     def _validate_and_initialize(self):
-        if anndata_version_is_pre_070() and self.server_config.adaptor__anndata_adaptor__backed:
+        if anndata_version_is_pre_070():
             warnings.warn(
-                "Use of --backed mode with anndata versions older than 0.7 will have serious "
-                "performance issues. Please update to at least anndata 0.7 or later."
+                "Use of anndata versions older than 0.7 will have serious "
```
Comment (MDunitz) [inline server/data_anndata/anndata_adaptor.py:182]:
```suggestion
                "Use of anndata versions older than 0.7 will have serious issues. Please update to at "
```

File: server/data_anndata/anndata_adaptor.py (hunk 3 above)
Comment (MDunitz) [inline server/data_anndata/anndata_adaptor.py:183]:
```suggestion
                "least anndata 0.7 or later."
```

**Pull request merged**